compiler/gentype/Paths.ml

```diff
@@ -29,17 +29,43 @@ let find_name_space cmt =
   cmt |> Filename.basename |> (Filename.chop_extension [@doesNotRaise])
   |> keep_after_dash
 
-let get_output_file_relative ~config cmt =
-  (cmt |> handle_namespace) ^ ModuleExtension.ts_input_file_suffix ~config
+let remove_project_root_from_absolute_path ~(config : Config.t) source_path =
+  let i = String.length config.project_root + 1 in
+  let n = String.length source_path - i in
+  (String.sub source_path i n [@doesNotRaise])
 
-let get_output_file ~(config : Config.t) cmt =
-  Filename.concat config.project_root (get_output_file_relative ~config cmt)
+let get_output_file_relative ~config source_path =
+  if Filename.is_relative source_path then
+    (source_path |> handle_namespace)
+    ^ ModuleExtension.ts_input_file_suffix ~config
+  else
```

Comment:
There's a little logic duplication in the if and else branches.
Perhaps unify so it's easier to maintain in future.